srcpkgs/giac/patches/malloc.patch

```diff
@@ -1,11 +1,11 @@
 --- a/src/History.cc	2017-10-02 10:23:46.000000000 +0200
 +++ b/src/History.cc	2017-11-14 15:42:12.119849965 +0100
-@@ -3828,7 +3828,7 @@
+@@ -4633,7 +4633,7 @@
        // mode_s += "Time: ";
        // double t=double(clock());
        // mode_s += xcas::print_DOUBLE_(t/CLOCKS_PER_SEC);
--#ifdef HAVE_MALLOC_H //
-+#if defined HAVE_MALLOC_H && defined __GLIBC__ //
+-#if defined(HAVE_MALLOC_H) && !defined(__MINGW_H)
++#if defined(HAVE_MALLOC_H) && !defined(__MINGW_H) && defined(__GLIBC__)
```

Comment:
They could just test for the function instead of the header >.<

Reply:
This means adding `AC_CHECK_FUNCS(mallinfo)` in `configure.ac` and replacing the test by `#ifdef HAVE_MALLINFO` or something like that?

I think the few cases where they check `HAVE_MALLOC_H` is just to use`mallinfo()`.

Is it worth doing that for us, or proposing it to upstream? It seems a chore having to deal with autoconf since the patch above works ok for us.

Reply:
Oh, really only for upstream. The patch is definitely good enough for our packaging :)